src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1946,13 +1925,21 @@ private ILabeledOperation CreateBoundLabeledStatementOperation(BoundLabeledState
 
         private IExpressionStatementOperation CreateBoundExpressionStatementOperation(BoundExpressionStatement boundExpressionStatement)
         {
-            IOperation expression = Create(boundExpressionStatement.Expression);
-            SyntaxNode syntax = boundExpressionStatement.Syntax;
-
             // lambda body can point to expression directly and binder can insert expression statement there. and end up statement pointing to
             // expression syntax node since there is no statement syntax node to point to. this will mark such one as implicit since it doesn't
             // actually exist in code
             bool isImplicit = boundExpressionStatement.WasCompilerGenerated || boundExpressionStatement.Syntax == boundExpressionStatement.Expression.Syntax;
+            SyntaxNode syntax = boundExpressionStatement.Syntax;
+
+            // If we're creating the tree for a speculatively-bound constructor initializer, there can be a bound sequence as the child node here
+            // that corresponds to the lifetime of any declared variables.
+            IOperation expression = Create(boundExpressionStatement.Expression);
+            if (boundExpressionStatement.Expression is BoundSequence sequence)
```

Comment:
>boundExpressionStatement.Expression is BoundSequence sequence [](start = 16, length = 61)

Are we getting Operation.None in scenario like that. It feels like we shouldn't be getting any for the speculation scenario. Did we get the None node before the change? If, so, we should open an issue to follow up and probably change Implicit to false on that node rather than on this one.

Reply:
> Are we getting Operation.None in scenario like that.

Yes, we're producing a NoneOperation, and we got it before the change. The verification I added to `Scope_ConstructorInitializers_06` actually asserts on master, because it also has the bug this is asserting on. The main issue is that we have 3 nodes here:
1. The outer expression statement, here. This has the ConstructorInitializerSyntax as it's syntax
2. The middle NoneOperation. The syntax of this node in the operation tree is the ArgumentListSyntax under the ConstructorInitializerSyntax.
3. The bottom invocation. This is the node with conflicting syntax, it has the ConstructorInitializerSyntax as well. While this node feels like the right node to be explicit for this tree because it corresponds to the actual user-typed code, I can look into making this one explicit instead.

---
In reply to: [527022462](https://github.com/dotnet/roslyn/pull/49453#discussion_r527022462) [](ancestors = 527022462)

Reply:
> Yes, we're producing a NoneOperation, and we got it before the change. 

We discussed this offline and, given that, the problem with  ```isImplicit``` was present before and it is not about conflict with the NoneOperation created for the BoundSequence, the current implementation makes sense.

---
In reply to: [527230162](https://github.com/dotnet/roslyn/pull/49453#discussion_r527230162) [](ancestors = 527230162,527022462)